### PR TITLE
Replace python's server module with flask.

### PR DESCRIPTION
**Pull request by neshdev**
Tests were failing on mac osx sporadically. Switching to a different web server seems to have solved the issues.

Added vscode files for development.

b/337248131

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -32,10 +32,21 @@ dependencies = [
 [tool.hatch.version]
 path = "src/kagglehub/__init__.py"
 
+[project.optional-dependencies]
+dev = [
```
Comment (rosbo) [inline pyproject.toml:36]: should this be added to `tool.hatch.envs.default` instead?

File: tests/test_model_upload.py [hunk 2]
```diff
@@ -1,246 +1,119 @@
...
+        stub.stop_server()
 
     def test_model_upload_with_invalid_handle(self) -> None:
-        with create_test_http_server(KaggleAPIHandler):
-            with self.assertRaises(ValueError):
-                with TemporaryDirectory() as temp_dir:
-                    test_filepath = Path(temp_dir) / TEMP_TEST_FILE
-                    test_filepath.touch()  # Create a temporary file in the temporary directory
-                    model_upload("invalid/invalid/invalid", temp_dir, APACHE_LICENSE, "model_type")
-                    self.assertEqual(len(KaggleAPIHandler.UPLOAD_BLOB_FILE_NAMES), 1)
```
Comment (rosbo) [inline tests/test_model_upload.py:131]: Why did you remove the assertions here?

File: tests/flask_stub_server.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    server.start()
+
+
+def stop_server() -> None:
+    global server  # noqa: PLW0602
+    server.shutdown()
+
+
+if __name__ == "__main__":
+    app.run("127.0.0.1", port=7777, debug=False, use_reloader=False)
```
Comment (rosbo) [inline tests/flask_stub_server.py:150]: Let's get the address and port from this environment variable: https://github.com/Kaggle/kagglehub/blob/9c25bbfe2bcfcce847cbb5ddbe080b607f63d153/tests/__init__.py#L7

This is the environment variable that configures the client to point to this test server.

File: tests/flask_stub_server.py [hunk 4]
```diff
@@ -0,0 +1,150 @@
+import threading
```
Comment (rosbo) [inline tests/flask_stub_server.py:1]: Are you planning to move the other tests to use flask?
Is the plan to add all the test cases to this shared stub server file? We will need to be careful to ensure this doesn't become a convoluted mess.

For instance, we have specific test setup for corner cases like these: https://github.com/Kaggle/kagglehub/blob/9c25bbfe2bcfcce847cbb5ddbe080b607f63d153/tests/test_kaggle_api_client.py#L22-L35

**Review by rosbo — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 1 above)
Comment (neshdev) [reply pyproject.toml:36]: I was doing local development on my mac as follows:

```
pip -m venv .env
./.env/bin/activate
pip install -e .[dev]
```

After, I can run all the commands locally: `ruff fmt`, `ruff lint`

File: tests/test_model_upload.py (hunk 2 above)
Comment (neshdev) [reply tests/test_model_upload.py:131]: The assertion is that this raises a ValueError. The other statements are not executed.

File: pyproject.toml (hunk 1 above)
Comment (rosbo) [reply pyproject.toml:36]: You should not have to create and manage virtual environment manually. `hatch` is handling all of that.

Comment (rosbo) [reply pyproject.toml:36]: And then, you can run the convenience script I created for fmt / lint etc: https://github.com/Kaggle/kagglehub?tab=readme-ov-file#lint--format

Comment (rosbo) [reply pyproject.toml:36]: The fmt script for instance calls `black`, `ruff` and `style`: https://github.com/Kaggle/kagglehub/blob/9c25bbfe2bcfcce847cbb5ddbe080b607f63d153/pyproject.toml#L70-L74

File: tests/flask_stub_server.py (hunk 3 above)
Comment (neshdev) [reply tests/flask_stub_server.py]: Done

File: tests/flask_stub_server.py (hunk 4 above)
Comment (neshdev) [reply tests/flask_stub_server.py]: As discussed offline, I will move all the tests over to flask. I will do a follow up PR.
